src/model/network-legacy.cpp

```diff
@@ -266,7 +266,8 @@ EvalStatus LegacyNetwork::ComputeAccesses(const tiling::CompoundTile& tile, cons
     // 1. link transfers should result in buffer accesses to a peer.
     // 2. should reductions via link transfers be counted as spatial or temporal?
     stats_.link_transfers[pv] = tile[pvi].link_transfers;
-    if (problem::GetShape()->IsReadWriteDataSpace.at(pv))
+    if (problem::GetShape()->IsReadWriteDataSpace.at(pv) &&
+            (specs_.cType & ConnectionType::UD) )
```

Comment:
Why aren't we checking the reduction flag here? In fact, I don't understand the semantics of a non-reducing Update operation if there's an address collision.